static/scripts/pay.ts

```diff
@@ -5,14 +5,86 @@ import { TxType, txData } from "./render-transaction";
 
 const permit2Address = "0x000000000022D473030F116dDEE9F6B43aC78BA3";
 
+const notifications = document.querySelector(".notifications") as Element;
+const claimButtonElem = document.getElementById("claimButton") as Element;
+const buttonMark = document.querySelector('.hide-load');
+const claimLoader = document.querySelector(".claim-loader") as Element;
+
+// Object containing details for different types of toasts
+const toastDetails = {
+    timer: 5000,
+    success: {
+        icon: 'fa-circle-check',
+    },
+    error: {
+        icon: 'fa-circle-xmark',
+    },
+    warning: {
+        icon: 'fa-triangle-exclamation',
+    },
+    info: {
+        icon: 'fa-circle-info',
+    }
+}
+
+const removeToast = (toast) => {
+    toast.classList.add("hide");
+    if(toast.timeoutId) clearTimeout(toast.timeoutId); // Clearing the timeout for the toast
+    setTimeout(() => toast.remove(), 500); // Removing the toast after 500ms
+}
+
+const createToast = (id: string, text: string) => {
+    // Getting the icon and text for the toast based on the id passed
+    const { icon } = toastDetails[id];
+    const toast = document.createElement("li"); // Creating a new 'li' element for the toast
+    toast.className = `toast ${id}`; // Setting the classes for the toast
+    // Setting the inner HTML for the toast
+    toast.innerHTML = `
+      <div class="column">
+          <i class="fa-solid ${icon}"></i>
+          <span>${text}</span>
+      </div>
+      <i class="fa-solid fa-xmark" onclick="removeToast(this.parentElement)"></i>
+    `;
+    notifications.appendChild(toast); // Append the toast to the notification ul
+    
+    // Setting a timeout to remove the toast after the specified duration
+    // @ts-ignore
+    toast!.timeoutId = setTimeout(() => removeToast(toast), toastDetails.timer);
+}
+
+const disableClaimButton = () => {
+  // @ts-ignore
+  claimButtonElem!.disabled = true;
+  
+  // @ts-ignore
+  claimLoader!.style!.display = 'block';
+  // @ts-ignore
+  buttonMark!.style!.display = 'none';
+}
+
+const enableClaimButton = () => {
+  // @ts-ignore
+  claimButtonElem!.disabled = false;
+  
+  // @ts-ignore
+  claimLoader!.style!.display = 'none';
+  // take it back to default
+  // @ts-ignore
```

Comment:
You should refrain from using unnecessary `@ts-ignore` as it defeats the whole purpose of typescript,
basic typescript knowledge is required.

Reply:
I use Typescript, but I just taught type checking that isn't necessary..

Its gonna have a style object regardless

Reply:
I think I'd just cast them as HTML Elements

Reply:
> I use Typescript, but I just taught type checking that isn't necessary..
> 
> Its gonna have a style object regardless

Type checking is necessary here,
but regardless, you shouldn't use `@ts-ignore`

Reply:
We should take the precommit hooks from our more mature repos (like ubiquity dollar) because I'm pretty certain we banned ts-ignore

Also use a class. Don't modify the style property directly. That doesn't scale well.

Reply:
Resolved it below